packages/@webex/plugin-presence/src/presence-worker.ts

```diff
@@ -0,0 +1,277 @@
+import {debounce} from 'lodash';
+
+import {IGenericKeyValue, IWebex} from './interface';
+import {
+  FETCH_DELAY,
+  GROUNDSKEEPER_INTERVAL,
+  SUBSCRIPTION_DELAY,
+  UPDATE_PRESENCE_DELAY,
+  EXPIRED_PRESENCE_TIME,
+  PREMATURE_EXPIRATION_SUBSCRIPTION_TIME,
+  DEFAULT_SUBSCRIPTION_TTL,
+  APHELEIA_SUBSCRIPTION_UPDATE,
+  PRESENCE_UPDATE,
+  ENVELOPE_TYPE,
+} from './constants';
+
+/**
+ * Presence Worker
+ * Manages fetches and subscriptions for presence
+ * @class
+ */
+export default class PresenceWorker {
+  private presences: IGenericKeyValue; // current presence objects; updated time
+  private watchers: IGenericKeyValue; // counter of visible presence-required objects
+  private fetchers: IGenericKeyValue; // waiting to get presence object
+  private flights: IGenericKeyValue; // in flight to get presence object
+  private campers: IGenericKeyValue; // waiting to subscribe; updated time
+  private subscribers: IGenericKeyValue; // current subscriptions; expiration time
+  private webex: IWebex;
+
+  /**
+   * Constructs a presence worker to execute and
+   * maintain tasks related to presence upkeep.
+   */
+  constructor() {
+    this.presences = {};
+    this.watchers = {};
+    this.fetchers = {};
+    this.flights = {};
+    this.campers = {};
+    this.subscribers = {};
+  }
+
+  /**
+   * Connect to the mercury for presence and starts worker.
+   * @param {object} webex
+   * @returns {undefined}
+   */
+  initialize(webex: IWebex): void {
```

Comment:
In the meetings plugin, we have just added `@ts-ignore` when referring the `webex` object.

Should we consider doing the same?

Reply:
In the `plugin-meetings`, it is a bit complex and since couldn't figure out the type at that moment we ignored the type, here it seemed possible to add. 

In the future, we should create a base Webex interface and extend it in each plugin based on what is being added.

Reply:
Makes sense. Thank you, @rarajes2